### PR TITLE
Allow busybox to provide ip/tc

### PR DESCRIPTION
**Pull request by CameronNemo**

### CODE REVIEW

File: srcpkgs/busybox/template [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Template file for 'busybox'
 pkgname=busybox
 version=1.30.1
```
Comment (ailiop-git) [inline srcpkgs/busybox/template]: how about also bumping busybox to the latest stable version (1.31.1)?

Comment (CameronNemo) [reply srcpkgs/busybox/template]: see #15841

Comment (CameronNemo) [reply srcpkgs/busybox/template]: #15841 is closed and busybox is at the latest version

**Comment by Piraty** [comment]
any reason this is still a draft?

**Comment by CameronNemo** [comment]
@Piraty xbps cannot handle the switch to an alternative.

**Comment by CameronNemo** [comment]
I will just wait until XBPS is ready.

**Comment by heliocat** [comment]
I wasn't able to replicate this failure using the above patches:

```
carbon:~/src/void-linux/void-packages/hostdir/binpkgs/testing$ sudo xbps-install -R . iproute2 busybox
Password: 

Name     Action    Version           New version            Download size
iproute2 update    5.11.0_1          5.11.0_2               - 
busybox  update    1.32.1_1          1.32.1_2               - 

Size required on disk:        3975KB
Space available on disk:        14GB

Do you want to continue? [Y/n] 

[*] Verifying package integrity
iproute2-5.11.0_2: verifying SHA256 hash...
busybox-1.32.1_2: verifying SHA256 hash...

[*] Collecting package files
iproute2-5.11.0_2: collecting files...
iproute2-5.11.0_1: collecting files...
busybox-1.32.1_2: collecting files...
busybox-1.32.1_1: collecting files...

[*] Unpacking packages
iproute2-5.11.0_1: updating to 5.11.0_2 ...
iproute2-5.11.0_2: unpacking ...
iproute2-5.11.0_2: registered 'ip' alternatives group
Creating 'ip' alternatives group symlink: ip -> /usr/bin/iproute2-ip
iproute2-5.11.0_2: registered 'tc' alternatives group
Creating 'tc' alternatives group symlink: tc -> /usr/bin/iproute2-tc
busybox-1.32.1_1: updating to 1.32.1_2 ...
Removing 'ash' alternatives group symlink: ash
busybox-1.32.1_2: unpacking ...
Creating 'ash' alternatives group symlink: ash -> /usr/bin/busybox
busybox-1.32.1_2: registered 'ip' alternatives group
busybox-1.32.1_2: registered 'tc' alternatives group

[*] Configuring unpacked packages
iproute2-5.11.0_2: configuring ...
iproute2-5.11.0_2: updated successfully.
busybox-1.32.1_2: configuring ...
busybox-1.32.1_2: updated successfully.

0 downloaded, 0 installed, 2 updated, 2 configured, 0 removed.
carbon:~/src/void-linux/void-packages/hostdir/binpkgs/testing$ readlink -f /usr/bin/ip
/usr/bin/iproute2-ip
```
```

**Comment by CameronNemo** [comment]
@heliocat you need to make this test case pass: https://github.com/void-linux/xbps/blob/4a5eb8dc87cf08ad919ec6602ef5a70e8514710c/tests/xbps/xbps-alternatives/main_test.sh#L849

Currently it is set to expect failure.

**Comment by heliocat** [comment]
Shouldn't `test "$(readlink -f root/usr/bin/file)" = "pkg-b-file"` be `test "$(readlink root/usr/bin/file)" = "pkg-b-file"`? `realpath`/`readlink -f` will fully canonicalize the path and return an absolute which won't then pass the string equality test. I'll give the test a better pass when I've got some more time but I think there are a few logic issues in there.

**Comment by heliocat** [comment]
OK, I found two problems with the test and one with my own testing methodology. The two test problems are: it shouldn't call readlink -f (noted above), and the casing on pkg-b-file is inconsistent My own testing (the one that successfully upgraded busybox and iproute2 to alternatives-providing versions) was flawed because I did iproute2 in the transaction first which replaced the owned file before installing a new alternative provider. Had I let ordered things alphabetically (or let xbps-install do it automatically) I would have seen the bug.

**Pull request closed without merging**